Formula/fish.rb

```diff
@@ -49,7 +52,24 @@ def post_install
     (pkgshare/"vendor_conf.d").mkpath
   end
 
+  def caveats
+    <<~EOS
+      #{Tty.bold}fish#{Tty.reset} has been installed to #{Tty.underline}#{HOMEBREW_PREFIX}/bin/fish#{Tty.reset}
+
+      To set fish as your default login shell:
+
+      #{Tty.bold}chsh -s #{HOMEBREW_PREFIX}/bin/fish#{Tty.reset}
+
+      If you get a "non-standard shell" error:
+
+      All shells must be listed in /etc/shells to permit use as a login shell.
+      So append the install list of paths as sudo:
+
+      #{Tty.bold}echo #{HOMEBREW_PREFIX}/bin/fish | sudo tee -a /etc/shells#{Tty.reset}
+    EOS
+  end
+
```

Comment:
Not a maintainer, but I seem to remember setup caveats like this being rejected simply because they're not unique to Homebrew. No Homebrew shell formula sports such caveats for that reason.

Reply:
Hm, well. If we explain this with cmake install's output, it won't be seen. We do show instructions exactly like this with our own os-native package.
<img width="732" alt="150664582-fcb261ac-5056-4bec-afbb-4f3e64466785" src="https://user-images.githubusercontent.com/291142/151686908-ae4a0105-ba12-48d9-9326-810eb5d3838e.png">



But I find this a bit homebrow-specific in that the path is kinda specific to homebrew. We're currently seeing some users surprised by homebrew changing from /usr/local, which also happens to be what our thing does on it's own, to /opt/homebrew on arm64. https://github.com/fish-shell/fish-shell/issues/8657

Reply:
The one bit that we do go forward with for our .pkg installer, is we add fish to /etc/shells as part of the installation. It's just not practical with homebrew, unless we are allowed to require sudo privileges to make changes in etc. So we give those manual instructions for that bit.

Reply:
This warning was present in the past but removed in 3f3cf646745c35c3b84a03bd7ad525e220a0b6f9 (#43107).